### PR TITLE
Add search description in the rentals endpoint

### PR DESCRIPTION
**Pull request by Fredar**

### CODE REVIEW

**Comment by Fredar** [comment]
image

File: content/reference/endpoints/rentals.md [hunk 1]
```diff
@@ -170,6 +170,43 @@ GET /rentals
...
+
+~~~
+GET /rentals/search?start_at=20140324&end_at=20140330
+~~~
+
+### Search Parameters
+
+Name              | Type    | Default | Description
+------------------|---------|--------------
+amenities         | String  |         | Rentals with given amenities, each amenity has to be a separate parameter.
```
Comment (ZenCocoon) [inline content/reference/endpoints/rentals.md]: This needs an example to be really clear.

File: content/reference/endpoints/rentals.md [hunk 2]
```diff
@@ -170,6 +170,43 @@ GET /rentals
...
+direction         | String  |         | Direction to sort rentals, either `asc` or `desc`.
+guests            | Integer |         | Rentals with maximum number of sleeps above given number of guests.
+include_tentative | Boolean | false   | Treat tentative periods as available ones.
+max_price         | Integer |         | Rentals with `max_price` below given price. Will use `initial_price` if computed.
+min_price         | Integer |         | Rentals with `min_price` above given price. Will use `initial_price` if computed.
+page              | Integer |         | Used to browse multi-paged listings.
+per_page          | Integer |         | Limit amount of returned rentals to given number. Full list can then be browsed with `page` parameter.
+rental_type       | String  |         | List of comma separated types, all types described in [enums section](/reference/enums#rental-types).
+sort              | String  |         | Sort rentals by given attribute. Possible values: `price`, `sleeps`.
+special_offers    | String  |         | Rentals with special offers.
```
Comment (ZenCocoon) [inline content/reference/endpoints/rentals.md]: This should be boolean no?

File: content/reference/endpoints/rentals.md [hunk 3]
```diff
@@ -170,6 +170,43 @@ GET /rentals
...
+amenities         | String  |         | Rentals with given amenities, each amenity has to be a separate parameter.
+bb_ne             | Float   |         | Rentals located between `bb_sw` & `bb_ne`. Format `latitude, longitude`.
+bb_sw             | Float   |         | Rentals located between `bb_sw` & `bb_ne`. Format `latitude, longitude`.
+city              | String  |         | Rentals matching given city.
+country_code      | String  |         | Rentals matching given country code, list of codes is available at [wikipedia](http://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#Officially_assigned_code_elements).
+destination       | String  |         | List of comma separated destinations.
+direction         | String  |         | Direction to sort rentals, either `asc` or `desc`.
+guests            | Integer |         | Rentals with maximum number of sleeps above given number of guests.
+include_tentative | Boolean | false   | Treat tentative periods as available ones.
+max_price         | Integer |         | Rentals with `max_price` below given price. Will use `initial_price` if computed.
```
Comment (ZenCocoon) [inline content/reference/endpoints/rentals.md]: Why based on `initial_price`? This should be based on `final_price`

File: responses/rentals/search-scope.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (ZenCocoon) [inline responses/rentals/search-scope.json]: Don't we return all by default and limited fields only if using `fields` ?

File: content/reference/endpoints/rentals.md (hunk 1 above)
Comment (Fredar) [inline content/reference/endpoints/rentals.md]: Indeed, just hoped we could with this before https://github.com/BookingSync/BookingSync/issues/608

File: content/reference/endpoints/rentals.md (hunk 2 above)
Comment (Fredar) [inline content/reference/endpoints/rentals.md]: Should be, but is not working this way. Proposing a change here as well https://github.com/BookingSync/BookingSync/issues/608. At the moment `special_offer=true` is the same as `special_offer=false` or `special_offer=foo`

File: content/reference/endpoints/rentals.md (hunk 3 above)
Comment (Fredar) [inline content/reference/endpoints/rentals.md]: Ok will change wording to `final`. Just for the record - both are the same as the moment of this comment.

File: responses/rentals/search-scope.json (hunk 4 above)
Comment (Fredar) [inline responses/rentals/search-scope.json]: If you mean attributes, then no - we use light rental serializer for search and thats all is there.

File: content/reference/endpoints/rentals.md (hunk 1 above)
Comment (ZenCocoon) [inline content/reference/endpoints/rentals.md]: I'm actually not sure it's the best idea, such way can allow us to expand further an support strict exclusion.

for instance, `&tv=-1` could allow to return only rentals that don't have TV

File: content/reference/endpoints/rentals.md (hunk 3 above)
Comment (ZenCocoon) [inline content/reference/endpoints/rentals.md]: Currently yes, `initial_price` is already the `final_price` We'll change this and include traces about how the price was computed.

File: responses/rentals/search-scope.json (hunk 4 above)
Comment (ZenCocoon) [inline responses/rentals/search-scope.json]: ok, we should at least add the `currency` then

Comment (Fredar) [inline responses/rentals/search-scope.json]: Added `currency`.

**Comment by Fredar** [comment]
What do you think about adding a full list of amenities to `enumerables` section?

**Comment by ZenCocoon** [comment]
While they are accessible from the `amenities` endpoint, that sound like a good idea.

**Comment by Fredar** [comment]
Added a new ticket for list of amenities.

**Pull request merged**